### PR TITLE
Add examples directory and basic kafka example

### PR DESCRIPTION
**Pull request by kahowell**

### CODE REVIEW

**Comment by kahowell** [comment]
@josejulio, would love your feedback here.

File: examples/quarkus-producer-consumer/src/main/java/org/acme/MockData.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    return OutgoingCloudEventMetadata.builder()
+        .withType("com.redhat.console.advisor.new-recommendations")
+        .withSource(URI.create("urn:redhat:source:console:app:advisor"))
+        .withId("urn:redhat:console:event:5864ac25-4c52-4c87-bd28-9909a4fa3187")
+        .withSubject("urn:redhat:subject:console:rhel:08e8ec2b-6a79-4f1d-bea4-a438da139493")
+        .withTimestamp(ZonedDateTime.now())
+        .withExtension("redhatorgid", "org123")
+        .withExtension("redhatconsolebundle", "rhel")
+        .withDataSchema(URI.create("https://console.redhat.com/api/schemas/apps/advisor/v1/advisor-recommendations.json"))
+        .build();
```
Comment (josejulio) [inline examples/quarkus-producer-consumer/src/main/java/org/acme/MockData.java:28]: Ah - so the event type class already exists and is only missing the extension data.
It makes sense - would the `data` be already considered valid? i.e. would some validation run outside of the applications to guarantee that?

Comment (kahowell) [reply examples/quarkus-producer-consumer/src/main/java/org/acme/MockData.java:28]: It's really up the consumer whether to validate or not, but thinking we add a validation example at some point.

**Comment by kahowell** [comment]
Going to self-merge this since still somewhat WIP.

**Pull request merged**